python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -171,21 +150,44 @@ def __enter__(self) -> "SensorEvaluationContext":
         return self
 
     def __exit__(self, *exc) -> None:
-        self._resources_cm.__exit__(*exc)  # pylint: disable=no-member
```

Comment:
i think it is fine that we lose this right? The idea before was that an exception might get raised while the context was in scope and we might want to give the resource contextmanager a chance to catch it?

Reply:
Is this not handled by the `ExitStack`'s `__exit__`? maybe we are losing behavior here

Reply:
the condition we need to handle at [1] is if we did not `__enter__` (`self._cm_scope_entered is False`) then we should run `__exit__` right before we raise the exception